File: src/main/java/com/jamieswhiteshirt/literalascension/core/plugin/LiteralAscensionPlugin.java
package com.jamieswhiteshirt.literalascension.core.plugin;

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;

import java.util.Map;

@IFMLLoadingPlugin.Name("literalascension")
@IFMLLoadingPlugin.MCVersion("1.10.2")
@IFMLLoadingPlugin.SortingIndex(1003)
public class LiteralAscensionPlugin implements IFMLLoadingPlugin {
    @Override
    public String getModContainerClass() {
        return null;
    }

    @Override
    public String[] getASMTransformerClass() {
        return new String[]{"com.jamieswhiteshirt.literalascension.core.patcher.LiteralAscensionRuntimePatcher"};
    }

    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) { }

    @Override
    public String getAccessTransformerClass() {
        return null;
    }
}

File: src/main/kotlin/com/jamieswhiteshirt/literalascension/core/plugin/LiteralAscensionPlugin.kt
package com.jamieswhiteshirt.literalascension.core.plugin

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin

@IFMLLoadingPlugin.Name("literalascension")
@IFMLLoadingPlugin.MCVersion("1.10.2")
@IFMLLoadingPlugin.SortingIndex(1003)
class LiteralAscensionPlugin : IFMLLoadingPlugin {
    override fun getModContainerClass(): String? = null

    override fun getASMTransformerClass(): Array<out String> = arrayOf("com.jamieswhiteshirt.literalascension.core.patcher.LiteralAscensionRuntimePatcher")

    override fun getSetupClass(): String? = null

    override fun injectData(data: MutableMap<String, Any>?) { }

    override fun getAccessTransformerClass(): String? = null
}